### PR TITLE
fix: recommend install CLIs using `--locked` flag

### PR DESCRIPTION
**Pull request by amrbashir**

### CODE REVIEW

File: .github/workflows/templates-test.yml [hunk 1]
```diff
@@ -96,19 +96,19 @@ jobs:
...
         if: (matrix.settings.manager == 'cargo' || matrix.settings.manager == 'dotnet') && matrix.settings.rc != true
 
-      - run: cargo install tauri-cli --version '^2.0.0-rc'
+      - run: cargo install tauri-cli --version '^2.0.0-rc' --locked
         if: (matrix.settings.manager == 'cargo' || matrix.settings.manager == 'dotnet') && matrix.settings.rc == true
 
       - run: |
           rustup target add wasm32-unknown-unknown
-          cargo install --locked trunk
+          cargo install --locked trunk --locked
```
Comment (amrbashir) [inline .github/workflows/templates-test.yml]:
```suggestion
          cargo install trunk --locked
```

File: .github/workflows/templates-test.yml [hunk 2]
```diff
@@ -96,19 +96,19 @@ jobs:
...
         if: matrix.settings.install_trunk
       - run: |
           rustup target add wasm32-unknown-unknown
-          cargo install --locked dioxus-cli
+          cargo install --locked dioxus-cli --locked
```
Comment (amrbashir) [inline .github/workflows/templates-test.yml]:
```suggestion
          cargo install dioxus-cli --locked
```

File: .github/workflows/audit.yml [hunk 3]
```diff
@@ -27,14 +27,15 @@ concurrency:
...
 jobs:
   audit-rust:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - uses: dtolnay/rust-toolchain@stable
-      - run: cargo install cargo-audit
-      - run: cargo audit
+      - uses: actions/checkout@v4
+      - uses: rustsec/audit-check@v1.4.1
```
Comment (FabianLars) [inline .github/workflows/audit.yml]: is there a reason not to use v2? It's just 3 days old so genuine question.

Comment (amrbashir) [reply .github/workflows/audit.yml]: I just copied from their README.md, didn't notice there was v2

Comment (amrbashir) [inline .github/workflows/audit.yml]:
```suggestion
      - uses: rustsec/audit-check@v2
```

File: .github/workflows/audit.yml [hunk 4]
```diff
@@ -27,14 +27,15 @@ concurrency:
...
+
 jobs:
   audit-rust:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - uses: dtolnay/rust-toolchain@stable
-      - run: cargo install cargo-audit
-      - run: cargo audit
+      - uses: actions/checkout@v4
```
Comment (Bratan613) [inline .github/workflows/audit.yml:35]:
```suggestion
      - uses: actions/checkout@v4
```

**Review by Bratan613 — COMMENTED** [review]
@tauri-bot

**Pull request merged**